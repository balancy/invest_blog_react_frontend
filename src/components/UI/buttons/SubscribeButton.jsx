import React from 'react'

const SubscribeButton = ({ children, ...props }) => {
    return (
        <button {...props} className="SubscribeButton">
            {children}
        </button>
    )
}

export default SubscribeButton
