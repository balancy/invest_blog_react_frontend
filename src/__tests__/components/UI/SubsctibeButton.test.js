import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SubscribeButton from '../../../components/UI/buttons/SubscribeButton';


test('should render SubscribeButton component', () => {
    render(
        <SubscribeButton
            data-testid='subButton-1'
            name='firstButton'
            disabled
        >
            Button text
        </SubscribeButton>
    )
    const subscribeButton = screen.getByTestId('subButton-1')

    expect(subscribeButton).toBeDisabled()
    expect(subscribeButton).toBeInstanceOf(HTMLButtonElement)
    expect(subscribeButton).toBeInTheDocument()
    expect(subscribeButton).toContainHTML('button')
    expect(subscribeButton).toHaveAttribute('name')
    expect(subscribeButton).toHaveClass('SubscribeButton')
    expect(subscribeButton).toHaveTextContent('Button text')
})


test('matches snapshot', () => {
    const tree = renderer
        .create(
            <SubscribeButton
                data-testid='subButton-1'
                name='firstButton'
                disabled
            >
                Button text
            </SubscribeButton>
        ).toJSON()
    expect(tree).toMatchSnapshot()
})