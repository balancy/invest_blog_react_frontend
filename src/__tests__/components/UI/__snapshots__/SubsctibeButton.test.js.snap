// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="SubscribeButton"
  data-testid="subButton-1"
  disabled={true}
  name="firstButton"
>
  Button text
</button>
`;
